x-pack/plugins/observability/public/components/threshold/threshold_rule_expression.tsx

```diff
@@ -430,34 +450,7 @@ export default function Expressions(props: Props) {
           />
         </EuiButtonEmpty>
       </div>
-      <EuiSpacer size={'m'} />
-      <EuiFormRow
-        label={i18n.translate('xpack.observability.threshold.rule.alertFlyout.filterLabel', {
-          defaultMessage: 'Filter (optional)',
-        })}
-        helpText={i18n.translate('xpack.observability.threshold.rule.alertFlyout.filterHelpText', {
-          defaultMessage: 'Use a KQL expression to limit the scope of your alert trigger.',
-        })}
-        fullWidth
-        display="rowCompressed"
-      >
-        {(metadata && derivedIndexPattern && (
-          <MetricsExplorerKueryBar
-            derivedIndexPattern={derivedIndexPattern}
-            onChange={debouncedOnFilterChange}
-            onSubmit={onFilterChange}
-            value={ruleParams.filterQueryText}
-          />
-        )) || (
-          <EuiFieldSearch
```

Comment:
I didn't see the [issue](https://github.com/elastic/kibana/issues/64012) that introduced this condition anymore.